### DISCUSSION TITLE
One validation for multiple number fields

### DISCUSSION THREAD

**Discussion by joeDH00** [Budibase/budibase#13383] [Help] [answered]
Hi, I am trying to validate multiple number fields. The idea is that there will be a state that stores a number. Lets say the number is 4. Now, theres 4 different number fields and I want the total number of those 4 number fields to be totalled up to 7. Is that possible.

![image](https://github.com/Budibase/budibase/assets/159232925/f74685c2-2847-407e-a39b-a876944db46c)

Field 1,2,3 and 4 must total up to 7 and return validation message unless the state is null in which case does not need to validate

**Comment by mikesealey** [comment, marked as answer]
Hi @joeDH00 

I've been looking in to this for you. Here's my solution:

* Put each of your fields into a container
* In the container settings, set `gap` to `none`
* Within each of those containers add a `Paragraph` component, and set it to your validation message (eg: "Sum of fields must equal 7"), apply styling (red, extra small
* On each of the `Paragraph` components, in the Conditions tab, set "Hide component if", _(sum of four fields)_*, is equal to 7.
* Optional condition - Hide if field is empty, or even Hide if all fields are empty.
* As always, styling is completely up to you. 


\* I used Javascript for this, you may find it simpler than using helpers. 

![image](https://github.com/Budibase/budibase/assets/39269380/a57523dc-120f-4c8b-9f04-d77979dfba99)

I've also attached an app-export that I used to work on this for you. Feel free to have a closer look...
[13383cloud-export-1713801197798.tar.gz](https://github.com/Budibase/budibase/files/15065479/13383cloud-export-1713801197798.tar.gz)
